README.md

```diff
@@ -131,83 +151,136 @@ const ipld = new Ipld({
+  - `remainderPath` (`string`): the part of the path that wasn’t resolved yet.
+  - `value` (`*`): the value where the resolved path points to. If further traversing is possible, then the value is a CID object linking to another IPLD Node. If it was possible to fully resolve the path, `value` is the value the `path` points to. So if you need the CID of the IPLD Node you’re currently at, just take the `value` of the previously returned IPLD Node.
+
+
+### `.get(cid)`
+
+> Retrieve an IPLD Node.
+
+ - `cid` (`CID`): the CID of the IPLD Node that should be retrieved.
+
+Returns a Promise with the IPLD Node that correspond to the given `cid`.
+
+Throws an error if the IPLD Node can’t be retrieved.
+
+
+### `.getMany(cids)`
+
+> Retrieve several IPLD Nodes at once.
+
+ - `cids` (`Iterable<CID>`): the CIDs of the IPLD Nodes that should be retrieved.
+
+Returns an async iterator with the IPLD Nodes that correspond to the given `cids`.
+
+Throws an error if a IPLD Node can’t be retrieved.
 
-> Store the given node of a recognized IPLD Format.
 
-`options` is an object that must contain one of the following combinations:
-- `cid` - the CID of the node
-- `[hashAlg]`, `[version]` and `format` - the hashAlg, version and the format that should be used to create the CID of the node. The
-`hashAlg` and `version` defaults to the default values for the `format`.
+### `.remove(cid)`
 
-It may contain any of the following:
+> Remove an IPLD Node by the given `cid`
 
-- `onlyHash` - If true the serialized form of the node will not be passed to the underlying block store but the passed callback will be invoked as if it had been
+ - `cid` (`CID`): the CIDs of the IPLD Node that should be removed.
 
-`callback` is a function that should have the signature as following: `function (err, cid) {}`, where `err` is an Error object in case of error and `cid` is the cid of the stored object.
+Throws an error if the IPLD Node can’t be removed.
 
-### `.get(cid [, path] [, options], callback)`
 
-> Retrieve a node by the given `cid` or `cid + path`
+### `.removeMany(cids)`
 
-`options` is an optional object containing:
+> Remove IPLD Nodes by the given `cids`
 
-- `localResolve: bool` - if true, get will only attempt to resolve the path locally
+ - `cids` (`Iterable<CID>`): the CIDs of the IPLD Nodes that should be removed.
 
-`callback` should be a function with the signature `function (err, result)`, the result being an object with:
+Throws an error if any of the Blocks can’t be removed. This operation is not atomic, some Blocks might have already been removed.
 
-- `value` - the value that resulted from the get
-- `remainderPath` - If it didn't manage to successfully resolve the whole path through or if simply the `localResolve` option was passed.
-- `cid` - Where the graph traversal finished - if `remainderPath` has a value, this will be where it has its root
 
-### `.getMany(cids, callback)`
+### `.tree(cid, [path], [options])`
```

Comment:
Sidenote: I find `tree` to be a confusing name especially since it just returns paths. I think `paths` or `list` would have being far more clear. 

I also kind of wish there was a way to traverse without following every branch   `.traverse(cid, path, {follow:path => boolean, order:path[] => path[] })` that way `traverse(cid, path)` is shallow, and `.traverse(cid, path, {follow:Boolean})` is recursive, but it also provides an opportunity to avoid traversing irrelevant paths. As of `order` it can provide a way to choose traversal scheme e.g. depth-first could be as simple as returning path with most path components first.

Reply:
Extracted into https://github.com/ipld/js-ipld/issues/198.